### ISSUE TITLE
Flaky-test: AdminApiTest.testNamespaceSplitBundleConcurrent

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#10788]
AdminApiTest flaky. The testNamespaceSplitBundleConcurrentis test method fails sporadically.

[example failure](https://github.com/apache/pulsar/pull/9927/checks?check_run_id=2729711384#step:10:2980)

```
Error:  testNamespaceSplitBundleConcurrent(org.apache.pulsar.broker.admin.AdminApiTest)  Time elapsed: 0.141 s  <<< FAILURE!
java.lang.AssertionError: split bundle shouldn't have thrown exception
	at org.testng.Assert.fail(Assert.java:99)
	at org.apache.pulsar.broker.admin.AdminApiTest.testNamespaceSplitBundleConcurrent(AdminApiTest.java:1466)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by lhotari** [comment]
previous similar flaky test issue: #9595

**Comment by 315157973** [comment]
Hello @lhotari , do we have a way to view or download the complete log at that time (including the Broker side)

It's hard to reproduce locally

**Comment by mattisonchao** [comment]
I want to fix this issue.

**Comment by lhotari** [comment]
Another failure [here](https://github.com/apache/pulsar/pull/11407/checks?check_run_id=3131354883#step:10:441):
```
Error:  Tests run: 68, Failures: 1, Errors: 0, Skipped: 36, Time elapsed: 208.968 s <<< FAILURE! - in org.apache.pulsar.broker.admin.v1.V1_AdminApiTest
Error:  testNamespaceSplitBundleConcurrent(org.apache.pulsar.broker.admin.v1.V1_AdminApiTest)  Time elapsed: 0.243 s  <<< FAILURE!
java.lang.AssertionError: expected [8] but found [7]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
	at org.testng.Assert.assertEquals(Assert.java:122)
	at org.testng.Assert.assertEquals(Assert.java:907)
	at org.testng.Assert.assertEquals(Assert.java:917)
	at org.apache.pulsar.broker.admin.v1.V1_AdminApiTest.testNamespaceSplitBundleConcurrent(V1_AdminApiTest.java:1096)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by lhotari** [comment]
> It's hard to reproduce locally

I didn't try reproducing this, but I have had success reproducing issues by running the tests inside docker with resource limits. This helps simulate the resource constraints that there is in GitHub Actions Runner VMs.
I have [shell scripts for this.](https://github.com/lhotari/pulsar-contributor-toolbox#ptbx_until_test_fails_in_docker) The solution has been only tested on Linux since I use Linux for development. Similar approach might work in a VM that uses Linux.

**Comment by codelipenghui** [comment]
The issue had no activity for 30 days, mark with Stale label.

**Comment by aloyszhang** [comment]
This issue may be resolved by https://github.com/apache/pulsar/pull/14565

**Comment by RobertIndie** [comment]
This issue still happens: https://github.com/apache/pulsar/runs/7130098037?check_suite_focus=true

**Comment by nicoloboschi** [comment]
Another time: https://github.com/apache/pulsar/runs/7320980102?check_suite_focus=true

**Comment by codelipenghui** [comment]
The flaky test should be fixed by #16612
